File: DSA/Vyom-Singhal/Day1/Day1-q1.cpp
#include<iostream>
using namespace std;

int main() {
    int n;
    cout << "Enter size of array : ";
    cin >> n;
    int *arr = new int [n];
    cout << "Enter elements : ";
    for(int i = 0; i < n; i++) {
        cin >> arr[i];
    }
    int day, month, count = 0;
    cout << "Enter day and month : ";
    cin >> day >> month;
    for(int i = 0; i <= n -month; i++) {
        int sum = 0;
        for(int j = i; j < i+month; j++) {
            sum += arr[j];
        }
        if(sum == day) count++;
    }
    cout << "Unique ways : " << count;
}

File: DSA/Aanidhay-Aggarwal/Day-1/Day1_q1.java
import java.util.Scanner;

public class CandyBar {
    public static int numWays(int[] candy, int day, int month) {
        int count = 0;
        int windowSum = 0;
        for (int i = 0; i < candy.length; i++) {
            windowSum += candy[i];
            if (i >= month) {
                windowSum -= candy[i - month];
            }
            if (i >= month - 1 && windowSum == day) {
                count++;
            }
        }
        return count;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the candy array (space-separated): ");
        String[] candyStr = scanner.nextLine().split(" ");
        int[] candy = new int[candyStr.length];
        for (int i = 0; i < candyStr.length; i++) {
            candy[i] = Integer.parseInt(candyStr[i]);
        }

        System.out.print("Enter Raju's birth day: ");
        int day = scanner.nextInt();

        System.out.print("Enter Raju's birth month: ");
        int month = scanner.nextInt();

        int result = numWays(candy, day, month);
        System.out.println("Number of ways to divide the bar: " + result);
    }
}
